daemon/api.go

```diff
@@ -801,11 +804,20 @@ func modeFlags(devMode, jailMode bool) (snapstate.Flags, error) {
 		if devMode {
 			return flags, errModeConflict
 		}
+		if classic {
+			return flags, errClassicJailmodeConflict
+		}
 		flags.JailMode = true
 	}
-	if devMode || devModeOS {
+	if devMode {
```

Comment:
I think I wanted to maintain the visual flow of this function. It's already somewhat noisy so I wanted to do checks specific to "modes" (devmode, jail mode" and look for classic inside each of them. I'd love if I could represent the overall logic cleaner though.